pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -186,14 +189,24 @@ func RunBuildChain(f *clientcmd.Factory, cmd *cobra.Command, args []string) erro
 		// Set the specified repo as the only one to output dependencies for
 		repos[repo] = tags
 	} else {
-		// Get all image repositories from build configurations
-		repos = getRepos(buildConfigList)
-
-		// Make sure from now on that the --all flag is true
-		// since we are building dependency trees for every
-		// image repository available either in the current
-		// namespace or in every namespace
-		all = true
+		// all the tags of all the imagestreams across all the namespaces
+		for _, ns := range namespaces {
```

Comment:
Correct

----- Original Message -----
> > @@ -186,14 +189,24 @@ func RunBuildChain(f *clientcmd.Factory, cmd
> > *cobra.Command, args []string) erro
> >  		// Set the specified repo as the only one to output dependencies for
> >  		repos[repo] = tags
> >  	} else {
> > -		// Get all image repositories from build configurations
> > -		repos = getRepos(buildConfigList)
> > -
> > -		// Make sure from now on that the --all flag is true
> > -		// since we are building dependency trees for every
> > -		// image repository available either in the current
> > -		// namespace or in every namespace
> > -		all = true
> > +		// all the tags of all the imagestreams across all the namespaces
> > +		for _, ns := range namespaces {
> 
> Image streams definitely can exist by themselves.
> 
> Sent from my iPhone
> 
> > On Apr 23, 2015, at 4:46 AM, Michail Kargakis <notifications@github.com>
> > wrote:
> > 
> > In pkg/cmd/experimental/buildchain/buildchain.go:
> > 
> > > @@ -186,14 +189,24 @@ func RunBuildChain(f *clientcmd.Factory, cmd
> > > *cobra.Command, args []string) erro
> > >  		// Set the specified repo as the only one to output dependencies for
> > >  		repos[repo] = tags
> > >  	} else {
> > > -		// Get all image repositories from build configurations
> > > -		repos = getRepos(buildConfigList)
> > > -
> > > -		// Make sure from now on that the --all flag is true
> > > -		// since we are building dependency trees for every
> > > -		// image repository available either in the current
> > > -		// namespace or in every namespace
> > > -		all = true
> > > +		// all the tags of all the imagestreams across all the namespaces
> > > +		for _, ns := range namespaces {
> > I would argue that it's better to extract our streams from the build
> > configurations instead of iterating over all available namespaces for two
> > reasons: 1) since we build dep trees based on ICTs, buildConfigs is where
> > we want to look for streams, plus we might end up with more streams in our
> > results than needed (can we have an imageStream around w/o a respective
> > buildConfig?). 2) Searching through buildConfigs seems cheaper (no REST
> > calls) vs GET streams for every namespace. Anyway, if you follow what you
> > are doing right now, you should probably remove getRepos since it's won't
> > be used anymore.
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> > 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1693/files#r28956035